client/main.lua

```diff
@@ -267,101 +288,44 @@ local function carjackVehicle(target)
-                                },
-                            }) then
-                                TriggerServerEvent('qb-vehiclekeys:server:AcquireVehicleKeys', plate)
-                            end
-                            isTakingKeys = false
-                        end
-                    elseif config.lockNPCDrivenCars then
-                        TriggerServerEvent('qb-vehiclekeys:server:setVehLockState', NetworkGetNetworkIdFromEntity(entering), 2)
-                    else
-                        TriggerServerEvent('qb-vehiclekeys:server:setVehLockState', NetworkGetNetworkIdFromEntity(entering), 1)
-                        TriggerServerEvent('qb-vehiclekeys:server:AcquireVehicleKeys', plate)
-
-                        --Make passengers flee
-                        local pedsInVehicle = getPedsInVehicle(entering)
-                        for i = 1, #pedsInVehicle do
-                            local pedInVehicle = pedsInVehicle[i]
-                            if pedInVehicle ~= GetPedInVehicleSeat(entering, -1) then
-                                makePedFlee(pedInVehicle)
-                            end
-                        end
-                    end
-                -- Parked car logic
-                elseif driver == 0 and not Entity(entering).state.isOpen and not hasKeys(plate) and not isTakingKeys and not Entity(entering).state.vehicleid then
-                    TriggerServerEvent('qb-vehiclekeys:server:setVehLockState', NetworkGetNetworkIdFromEntity(entering), config.lockNPCParkedCars and 2 or 1)
-                end
-            end
-
-            -- Hotwiring while in vehicle, also keeps engine off for vehicles you don't own keys to
-            if cache.vehicle and not isHotwiring then
-                sleep = 1000
-                local plate = qbx.getVehiclePlate(cache.vehicle)
-                if cache.seat == -1
-                    and not hasKeys(plate)
-                    and not isBlacklistedVehicle(cache.vehicle)
-                    and not areKeysJobShared(cache.vehicle)
-                then
-                    sleep = 0
-                    local vehiclePos = GetOffsetFromEntityInWorldCoords(cache.vehicle, 0.0, 1.0, 0.5)
-                    qbx.drawText3d({ text = locale('info.search_keys'), coords = vehiclePos })
-                    SetVehicleEngineOn(cache.vehicle, false, false, true)
-
-                    if IsControlJustPressed(0, 74) then
-                        findKeys(cache.vehicle, plate)
-                    end
-                end
-            end
-
-            if config.carjackEnable and canCarjack then
-                local aiming, target = GetEntityPlayerIsFreeAimingAt(cache.playerId)
-                if aiming and target and target ~= 0 then
-                    if DoesEntityExist(target) and IsPedInAnyVehicle(target, false) and not IsEntityDead(target) and not IsPedAPlayer(target) then
+if config.carjackEnable then
+    CreateThread(function()
```

Comment:
How about we tie this thread to being created only when a player pulls a weapon and it valid for carjacking.

Something similar to this could be adapted
```lua
local function startWeaponStressThread(weapon)
    if isWhitelistedWeaponStress(weapon) then return end
    hasWeapon = true

    CreateThread(function()
        while hasWeapon do
            if IsPedShooting(cache.ped) then
                if math.random() <= config.stress.chance then
                    TriggerServerEvent('hud:server:GainStress', math.random(1, 5))
                end
            end
            Wait(0)
        end
    end)
end

AddEventHandler('ox_inventory:currentWeapon', function(currentWeapon)
    hasWeapon = false
    Wait(0)

    if not currentWeapon then return end

    startWeaponStressThread(currentWeapon.name)
end)
```

Reply:
Sounds good, need to remember that the script can be restarted when the player has a weapon in hand.

Reply:
well thats an edge case that we could have an onresourcestart and check cache.weapon is not false then trigger the loop.

Reply:
or perhaps just have `local hasWeapon = cache.weapon` at the top for reloads

Reply:
done